### ISSUE TITLE
Incorrect CS8911 for 'global using unsafe' with pointer type

### ISSUE THREAD

**Issue by Sergio0694** [dotnet/roslyn#77389]
**Version Used**: 4.13.0-3.25105.4 (400efa2d) (and tested both VS 17.13.0 and VS 17.14 P1)

**Steps to Reproduce**:

1. Checkout https://github.com/microsoft/CsWinRT/commit/443d0c47f1b6e6a16772035bc285b84e69ed4a1a
2. Open `src\cswinrt.sln`
3. Build `WinRT.Runtime2.csproj`

Ignore errors from other projects, just build this one project (it doesn't reference anything else).

**Diagnostic Id**: CS8911

**Expected Behavior**:

The code should compile just fine.

**Actual Behavior**:

The compiler produces an incorrect error:

![Image](https://github.com/user-attachments/assets/9070d73c-0927-46a9-88f7-2d64ae0f0d67)

cc. @CyrusNajmabadi

**Comment by Sergio0694** [comment]
It seems this is related to function pointers in particular:

```csharp
// This fails to build
global using unsafe PFNCONTEXTCALL = delegate*<void>;

// This builds fine
global using unsafe PFNCONTEXTCALL = void*;
```

**Comment by 333fred** [comment]
I attempted to reproduce this in a simple fashion by just testing just `global using unsafe X = delegate*<void>;` on the latest `main`, which does not issue any errors, nor does sharplab on 4.12. @Sergio0694, does this still reproduce for you? If so, can you please provide an isolated repro? That project does not appear to be generally buildable without a lot of setup.

**Comment by 333fred** [comment]
@tmat, this is happening because we're trying to encode a function pointer type for a portable PDB. It's trying to encode the using scope for a function pointer and, since [that does not support function pointers](https://github.com/dotnet/roslyn/blob/main/src/Compilers/Core/Portable/PEWriter/TypeNameSerializer.cs#L81-L89), it's giving an error. Any ideas on what to do here?
